teps/0090-matrix.md

```diff
@@ -596,17 +656,1130 @@ For example:
 
 Read more in the [documentation][ansible].
 
+## Proposal
+
+This proposal focuses on enabling execution a `PipelineTask` with different combinations
+of `Parameters`. This section will provide an overview, see the [design](#design) section
+below for further details.
+
+### API Change
+
+To support fanning out of `Tasks` in `Pipelines`, we propose adding a `Matrix` field to the
+`PipelineTask` specification that will be used to declare `Parameters` of type `Array`.
+
+```go
+type PipelineTask struct {
+	Name        string          `json:"name,omitempty"`
+	TaskRef     *TaskRef        `json:"taskRef,omitempty"`
+	TaskSpec    *EmbeddedTask   `json:"taskSpec,omitempty"`
+	Params      []Param         `json:"params,omitempty"`
+	Matrix      []Param         `json:"matrix,omitempty"`
+	...
+}
+```
+
+###### Alternatives
+* [Boolean in Parameter Specification](#api-change-boolean-in-parameter-specification)
+* [Array of Parameter Names in PipelineTask Specification](#api-change-array-of-parameter-names-in-pipelinetask-specification)
+
+### Fan Out
+
+The `Matrix` will be used to execute the `PipelineTask` in parallel `TaskRuns` or `Runs` with
+substitutions from combinations of the `Parameters` in the `Matrix`.
+
+The `Parameters` in the `Matrix` can use `Results` from previous `TaskRuns` or `Runs` to 
+dynamically generate `TaskRuns` or `Runs` from a given `PipelineTask` - see [details](#results).
+
+### Concurrency Control
+
+To support configuring the maximum number of `TaskRuns` or `Runs` generated from `Matrices`, we
+propose adding a field - `default-maximum-matrix-fan-out` - to [config defaults][config-defaults]
+with a default value of 256. Users can set it to a different value for their own Tekton Pipelines
+installations, similarly to other [installation customizations][custom-install], such as: 
```

Comment:
As discussed on Slack, updated the proposal:

> When a `Matrix` in `PipelineTask` would generate more than the maximum `TaskRuns` or `Runs`, this would fail the `Pipeline` in the first iteration. After initial usage of `Matrix`, we can explore other ways of supporting usage beyond that limit, such as allowing `TaskRuns` or `Runs` only up to the limit to run at a time, in a follow-up TEP.